CONTRIBUTING.md

```diff
@@ -1,38 +1,64 @@
 
-Issues are also a good place to present experience reports or requests for new
-features.
+Design principles:
 
-If you'd like to make changes to Twirp, read on:
+ * Simplicity. Keep serialization and routing simple and intuitive.
+ * Small api and interface. Reduce future support burden.
+ * Prevent user errors. Avoid surprising behavior.
+ * Pragmatism over being bleeding edge.
+ * Hooks are for observability instead of control flow, so it is easier to debug Twirp systems.
+ * No flags/options for code generation, so generated code is predictable across versions and platforms.
+ * As few dependencies as possible, so it is easier to integrate and upgrade.
+ * Prefer generated code over shared libraries between services and clients, so it is easier to implement changes without breaking compatibility.
 
-## Setup Requirements ##
+Contributions that are welcome:
 
-You will need git, Go 1.9+, and Python 2.7 installed and on your system's path.
-Install them however you feel.
+ * Security updates.
+ * Performance improvements.
+ * Supporting new versions of Golang and Protobug.
+ * Documentation.
+ * Making Twirp easier to integrate with other useful tools.
 
-## Developer Loop ##
+In the other hand, contributions that contradict the following priorities will be more difficult:
 
-Generally you want to make changes and run `make`, which will install all
-dependencies we know about, build the core, and run all of the tests that we
-have against all of the languages we support.
+ * Backwards compatibility is very important. Changes that break compatibility will see resistance, even when they only break a small use case. There must be a good reason behind a major version update.
+ * Complex and involved features should be avoided. Twirp was designed to be easy to use and easy to maintain. See the streaming proposal as an example of a complex feature that was not able to make it into a major release: https://github.com/twitchtv/twirp/issues/3.
+ * Features that could be implemented in a separate library should go in a separate library. Twirp allows easy integrtion with 3rd party code.
+
+
+## Report an Issue
+
+If you have run into a bug or want to discuss a new feature, please [file an issue](https://github.com/twitchtv/twirp/issues). If you'd rather not publicly discuss the issue, please email security@twitch.tv.
+
+## Contributing Code with Pull Requests
+
+Twirp uses github pull requests. Fork, hack away at your changes and submit. Most pull requests will go through a few iterations before they get merged. Different contributors will sometimes have different opinions, and often patches will need to be revised before they can get merged.
+
+### Requirements
 
-Most tests of the Go server are in `internal/twirptest/service_test.go`. Tests
-of cross-language clients are in the [clientcompat](./clientcompat) directory.
+ * Twirp officially supports the last 3 releases of Go.
+ * The Python implementation uses Python 2.7.
```

Comment:
Python has stopped official support for `2.7`, maybe remove the requirement?

https://github.com/python/devguide/pull/344

Reply:
That is right. However this should be discussed in a different PR. The Python implementation has a few things that should be discussed all together. This PR adds some edits to the Contributing Guidelines, but with the focus on adding the Twirp Principles.

Reply:
We edited this point with: "As such, it is not suitable for production use in its current form.". Thanks for the reference.